static/app/views/performance/transactionSummary/transactionSpans/spanDetails/utils.tsx

```diff
@@ -9,7 +9,8 @@ export function generateSpanDetailsRoute({
   orgSlug: String;
   spanSlug: SpanSlug;
 }): string {
-  return `/organizations/${orgSlug}/performance/summary/spans/${spanSlug.op}:${spanSlug.group}/`;
+  const spanComponent = `${encodeURIComponent(spanSlug.op)}:${spanSlug.group}`;
```

Comment:
I highly suggest not using `encodeURIComponent()` like this. `encodeURIComponent()` should be used for query strings, not path components of the URLs.


See something like: https://stackoverflow.com/questions/35947065/how-to-encode-for-use-in-a-url-path

Reply:
Can `spanSlug.op` be stripped like how the question are encoded into stackoverflow URLs?

They're not important for SO, but useful to be human readable. like https://stackoverflow.com/questions/35947065/i-am-tony-stark

Reply:
We already do stuff like this for release URLs see [example](https://github.com/getsentry/sentry/blob/master/static/app/views/eventsV2/table/tableView.tsx#L395-L404). But if this is not a good solution, one alternative is to change up the route for this view by encoding the op in the query string instead.

Reply:
Interesting 🤔 . We can give this a shot until it'sproblematicc